plugin/core/registry.py

```diff
@@ -95,11 +95,39 @@ def sessions(self, capability_path: Optional[str] = None) -> Generator[Session,
                     yield sv.session
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
-    def run(self, edit: Any) -> None:
+class LspRestartServerCommand(LspTextCommand):
+
+    ALL_SERVERS = 'All Servers'
+
+    def run(self, edit: Any, config_name: str = None) -> None:
         window = self.view.window()
-        if window:
-            windows.lookup(window).restart_sessions_async()
+        if not window:
+            return
+        self._config_names = [session.config.name for session in self.sessions()] if not config_name else [config_name]
+        if not self._config_names:
+            return
+        self._wm = windows.lookup(window)
+        if len(self._config_names) == 1:
+            self.restart_server(0)
+        else:
+            self._config_names.insert(0, self.ALL_SERVERS)
+            window.show_quick_panel(self._config_names, self.restart_server)
+
+    def restart_server(self, index: int) -> None:
+        if index < 0:
+            return
+
+        def run_async() -> None:
+            config_name = self._config_names[index]
+            if not config_name or config_name != self.ALL_SERVERS:
```

Comment:
I'm thinking that the "All servers" option is a bit inconsistent because when there are active servers in the view, it will be available and restart **all** servers, even the ones that are not active on the page.

On the other hand, if there is only one server or none, this option won't even show up.

Should this option:
 - only restart all **active** servers in the file
 - or be always available, as long as there are any active servers somewhere?
 - not be provided at all?

The last option would be the most consistent probably. The other would require some more non-trivial changes.

Reply:
Well, it's not common to restart all services, I agree